pallets/committee/src/lib.rs

```diff
@@ -588,6 +567,49 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
         <Voting<T, I>>::remove(&proposal);
         <Proposals<T, I>>::mutate(|proposals| proposals.retain(|h| h != &proposal));
     }
+
+    fn vote_referendum<F>(
+        origin: <T as frame_system::Trait>::Origin,
+        id: PipId,
+        call_maker: F,
+    ) -> DispatchResult
+    where
+        F: Fn() -> (<T as Trait<I>>::Proposal, <T as Trait<I>>::Proposal),
+    {
+        // Only committee members can use this function.
+        let who_key = AccountKey::try_from(ensure_signed(origin.clone())?.encode())?;
+        let who_id = Context::current_identity_or::<Identity<T>>(&who_key)?;
+        ensure!(Self::is_member(&who_id), Error::<T, I>::BadOrigin);
+
+        ensure!(
+            T::EnactProposalMaker::is_pip_id_valid(id),
+            Error::<T, I>::NoSuchProposal
+        );
+
+        // It creates the proposal if it does not exists or vote that proposal.
+        // let enact_call = T::EnactProposalMaker::enact_referendum_call(id);
+        let (aye_call, nay_call) = call_maker();
+        let hash = T::Hashing::hash_of(&aye_call);
+        // Self::deposit_event( RawEvent::VoteEnactReferendum(who_id, id));
+
+        if let Some(voting) = Self::voting(hash) {
+            Self::vote(origin, hash, voting.index, true)?;
+        } else {
+            Self::propose(origin, Box::new(aye_call))?;
+        }
+
+        // If voting info has been removed, the vote has finished.
+        // Then we have to clean Pip & the other call.
+        if !<Voting<T, I>>::contains_key(hash) {
```

Comment:
I think the logic for this part (lines 603 - 609) should be:
let nay_hash = T::Hashing::hash_of(&nay_call);
if Voting::contains_key(nay_hash) {
  if origin has a postive vote for nay_hash {
    delete origins positive vote for nay_hash
  }
}

Since we are enforcing that only positive votes are allowed, we don't need to worry about negative votes appearing.

Reply:
That code is not for a single vote. It is executed only when the proposal is accepted (ayes win) or rejected (nayes win).